include/LeMonADE/updater/UpdaterSimpleConnection.h

```diff
@@ -0,0 +1,176 @@
+   *
+   * @details It´s a virtual function for inheritance.
+   * Use this function for cleaning tasks (e.g. destroying arrays, file outut)
+   *
+   **/
+  virtual void cleanup(){};
+
+private:
+  //! Specialized move to be used for the movement of the monomers 
+  MoveType move;
+  //! 
+  ConnectionMoveType connectionMove;
+  //! Number of mcs to be executed
+  uint32_t nsteps;
+  //! random number generator (seed set in main program)
+  RandomNumberGenerators rng;
+  //! 
+  IngredientsType& ingredients;
+  //! number of monomers which could have a reaction bond
+  uint32_t NReactiveSites;
+  //! number of already reacted monomers
+  uint32_t NReactedSites;
+  //! get conversion of the reaction process
+  double getConversion(){return (1.*NReactedSites)/(1.*NReactiveSites);};
+  
+};
+/**Implementation of the member functions
+ * @brief 
+ */
+template<class IngredientsType,class MoveType, class ConnectionMoveType>
+bool UpdaterSimpleConnection<IngredientsType,MoveType,ConnectionMoveType>::execute()
+{
+  
+      time_t startTimer = time(NULL); //in seconds
+      std::cout<<"connection mcs "<<ingredients.getMolecules().getAge() << " passed time " << ((difftime(time(NULL), startTimer)) ) <<std::endl;
+      for(int n=0;n<nsteps;n++)
+      {
+	  std::vector<uint32_t> TagedMonomers; 
+	for(size_t m=0;m<ingredients.getMolecules().size();m++)
+	{
+		if( ingredients.getMolecules()[m].IsReactive() ) TagedMonomers.push_back(m);
+		move.init(ingredients);
+		if(move.check(ingredients)==true)
+		{
+			move.apply(ingredients);
+		}
+	}
+	uint32_t nTags(TagedMonomers.size());
+	
+	for(uint32_t i =0; i < nTags;i++)
+	{
+		// choose a random monomer 
+		uint32_t ID(TagedMonomers[ rng.r250_rand32() % nTags ]);
+		//vicinity reaction  
+		connectionMove.init(ingredients,ID);
+		if ( connectionMove.check(ingredients) )
+		{
+		  connectionMove.apply(ingredients);
+		  NReactedSites+=2;
+		}
+	}
+	ingredients.modifyMolecules().setAge(ingredients.getMolecules().getAge()+1);
+      }
+      std::cout <<"Conversion at "<<ingredients.getMolecules().getAge() << " is " << getConversion()  << std::endl;
+      std::cout<<"connection mcs "<<ingredients.getMolecules().getAge() << " with " << (((1.0*nsteps)*ingredients.getMolecules().size())/(difftime(time(NULL), startTimer)) ) << " [attempted connections/s]" <<std::endl;
+      std::cout<<"connection mcs "<<ingredients.getMolecules().getAge() << " passed time " << ((difftime(time(NULL), startTimer)) ) << " with " << nsteps << "connection MCS "<<std::endl;
+      return false;
+};
+template<class IngredientsType,class MoveType, class ConnectionMoveType>
+void  UpdaterSimpleConnection<IngredientsType,MoveType,ConnectionMoveType>::initialize()
+{
+  for(uint32_t i = 0 ; i < ingredients.getMolecules().size(); i++ )
+  {
+    if ( ingredients.getMolecules()[i].IsReactive() )
+    {
+      uint32_t NLinks(ingredients.getMolecules().getNumLinks(i));
+      uint32_t nIrreversibleBonds;
```

Comment:
**LOGICAL ERROR** forgotten initialization  
should read 
`uint32_t nIrreversibleBonds = 0`